services/campaigns/campaigns.go

```diff
@@ -0,0 +1,112 @@
+package campaigns
+
+import (
+	"bytes"
+	"encoding/json"
+
+	"github.com/cbroglie/mustache"
+	"github.com/google/uuid"
+
+	"github.com/mailbadger/app/entities"
+	"github.com/mailbadger/app/queue"
+	"github.com/mailbadger/app/storage"
+)
+
+type Service interface {
+	PrepareSubscriberEmailData(
+		s entities.Subscriber,
+		msg entities.SendCampaignParams,
+		campaign entities.Campaign,
+		html *mustache.Template,
+		sub *mustache.Template,
+		text *mustache.Template,
+		htmlBuf bytes.Buffer,
+		subBuf bytes.Buffer,
+		textBuf bytes.Buffer,
+	) (*entities.SendEmailTopicParams, error)
+	PublishSubscriberEmailParams(params *entities.SendEmailTopicParams) error
+}
+
+// service implements the Service interface
+type service struct {
+	db storage.Storage
+	p  queue.Producer
+}
+
+func New(db storage.Storage, p queue.Producer) Service {
+	return &service{
+		db: db,
+		p:  p,
+	}
+}
+
+func (svc *service) PrepareSubscriberEmailData(
+	s entities.Subscriber,
+	msg entities.SendCampaignParams,
+	campaign entities.Campaign,
```

Comment:
Instead of receiving a whole object of campaign, it's better to receive the campaign id since that's the only thing we're using here.